src/erlcloud_lambda.erl

```diff
@@ -0,0 +1,736 @@
+%%
+%%------------------------------------------------------------------------------
+-spec update_function_configuration(FunctionName :: binary(),
+                                    Description  :: binary() | undefined,
+                                    Handler      :: binary() | undefined,
+                                    MemorySize   :: integer() | undefined,
+                                    Role         :: binary() | undefined,
+                                    Timeout      :: 1..300 | undefined) -> return_val().
+update_function_configuration(FunctionName, Description,
+                              Handler, MemorySize, Role, Timeout) ->
+    update_function_configuration(FunctionName, Description, Handler,
+                                  MemorySize, Role, Timeout, default_config()).
+
+
+-spec update_function_configuration(FunctionName :: binary(),
+                                    Description  :: binary() | undefined,
+                                    Handler      :: binary() | undefined,
+                                    MemorySize   :: integer() | undefined,
+                                    Role         :: binary() | undefined,
+                                    Timeout      :: 1..300 | undefined,
+                                    Config       :: aws_config()) -> return_val().
+update_function_configuration(FunctionName, Description, Handler,
+                              MemorySize, Role, Timeout, Config) ->
+    Path = base_path() ++ "functions/" ++ binary_to_list(FunctionName) ++ "/configuration",
+    Json = filter_undef([{<<"Description">>, Description},
+                         {<<"Handler">>, Handler},
+                         {<<"MemorySize">>, MemorySize},
+                         {<<"Role">>, Role},
+                         {<<"Timeout">>, Timeout}]),
+    lambda_request(Config, put, Path, Json).
+%%------------------------------------------------------------------------------
+%% Utility Functions
+%%------------------------------------------------------------------------------
+
+from_record(#erlcloud_lambda_code{s3Bucket        = S3Bucket,
+                                  s3Key           = S3Key,
+                                  s3ObjectVersion = S3ObjectVersion,
+                                  zipFile         = ZipFile}) ->
+    List = [{<<"S3Bucket">>, S3Bucket},
+            {<<"S3Key">>, S3Key},
+            {<<"S3ObjectVersion">>, S3ObjectVersion},
+            {<<"ZipFile">>, ZipFile}],
+    filter_undef(List).
+
+filter_undef(List) ->
+    lists:filter(fun({_Name, Value}) -> Value =/= undefined end, List).
+
+base_path() ->
+    "/" ++ ?API_VERSION ++ "/".
+
+lambda_request(Config, Method, Path, Body) ->
+    lambda_request(Config, Method, Path, Body, []).
+
+lambda_request(Config, Method, Path, Body, QParam) ->
+    case erlcloud_aws:update_config(Config) of
+        {ok, Config1} ->
+            lambda_request_no_update(Config1, Method, Path, Body, QParam);
```

Comment:
because of the these method require data to be passed in the body of the request which is not supported in `erlcloud_aws:aws_request4`.  `erlcloud_retry:request` does most of the heavy lifting so theres not much duplication here